nengo/conftest.py

```diff
@@ -18,8 +19,7 @@
 
 
 def pytest_configure(config):
-    rc.reload_rc([])
-    rc.set('decoder_cache', 'enabled', 'False')
+    matplotlib.use('agg')
```

Comment:
> I consider it the user's job to set up their matplotlib properly.

My matplotlib is properly setup on all my systems. Afaik there is no way to tell tox to install additional modules into the venvs it creates given specific matplotlib backends. Also there is currently no way to solve the framework install issue with tox. So the only two options are: Set a defined default backend or require the user to always edit the config to run tests with `tox`.

> Sometimes, when I'm testing, I use plt.show to actually pop up a plot window.

That might no longer work with this because 'agg' is not an interactive renderer (i.e. can only plot to a file).



> I would like to be pretty sure that 'agg' will work on almost all systems for file output,

The matplotlib documentation says "The canonical renderer for user interfaces is Agg which uses the Anti-Grain Geometry C++ library to make a raster (pixel) image of the figure". That's not about file output, but at least better than nothing

I would also be ok with changing the backend to 'pdf' instead of 'agg' because we save PDFs anyways.